### PR TITLE
Feat: Adds tokens to `Price`

### PR DESCRIPTION
**Pull request by eduardoformiga**
## What's the purpose of this pull request?

This PR applies the new [Theming structure](https://github.com/vtex-sites/base.store/pull/407) to `Price`: new scoped tokens linked to global ones.

## How does it work?

This PR uses local variables to parameterize the `Price` properties, then connects these scoped tokens to the global ones.

#### Listing

| Local token                                          | Default value/Global token linked                                                                        |
|-|-|
| <code>--fs-price-listing-color</code>                | <code>var(--fs-color-text-light)</code> |
| <code>--fs-price-listing-text-decoration</code> | <code>line-through</code> |

#### Spot

| Local token                              | Default value/Global token linked       |
|-|-|
| <code>--fs-price-spot-margin-left</code> | <code>var(--fs-spacing-1)</code>        |
| <code>--fs-price-spot-font-weight</code> | <code>var(--fs-text-weight-bold)</code> |

## How to test it?

The `Price` component and its variants should look just as it was before these changes.

## Checklist

<em>You may erase this after checking them all ;)</em>

- [x] Added an entry in the `CHANGELOG.md` at the beginning of its due section. [The latest version should comes first](https://keepachangelog.com/en/1.0.0/#:~:text=The%20latest%20version%20comes%20first.).
- [x] Added the PR number with the PR link at the entry in the `CHANGELOG.md`. E.g., *New items in the `pull_request_template.md` ([#12](https://github.com/vtex-sites/gatsby.store/pull/12))* 

- PR description
- [x] Updated the Storybook - *if applicable*.
- [x] Added a label according to the PR goal - `Breaking change`, `Enhancement`, `Bug` or `Chore`.
- [x] Added the component, hook, or pathname in-between backticks (``) *- If applicable*. E.g., *`ComponentName` component*.
- [x] Identified the function or parameter in the PR *- If applicable*. E.g., *`useWindowDimensions` hook*.
- [x] For documentation changes, ping @ carolinamenezes, @ PedroAntunesCosta or @ Mariana-Caetano to review and update the changes.

### CODE REVIEW

File: src/components/ui/Price/Price.tsx [hunk 1]
```diff
@@ -15,11 +16,11 @@ type Props = PriceProps & {
   classes?: string
 }
 
-function Price({ classes, SRText, ...props }: Props) {
+function Price({ classes = '', SRText, ...props }: Props) {
   return (
     <>
       <SROnly text={SRText} />
-      <UIPrice className={`price ${classes}`} {...props} />
+      <UIPrice className={`${styles.fsPrice} ${classes}`} {...props} />
```
Comment (renatamottam) [inline src/components/ui/Price/Price.tsx:23]:
```suggestion
      <UIPrice data-fs-price className={`${styles.fsPrice} ${classes}`} {...props} />
```

File: src/components/sections/ProductDetails/product-details.scss [hunk 2]
```diff
@@ -50,7 +50,7 @@
 
   [data-store-buy-button] { width: 100%; }
 
-  .price[data-store-price] {
+  [data-store-price] {
```
Comment (renatamottam) [inline src/components/sections/ProductDetails/product-details.scss:53]:
```suggestion
  [data-fs-price] {
```

File: src/components/ui/Price/price.module.scss [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  // Design Tokens for Price
+  // --------------------------------------------------------
+
+  // Listing
+  --fs-price-listing-color: var(--fs-color-text-light);
+  --fs-price-listing-text-decoration-line: line-through;
+
+  // Spot
+  --fs-price-spot-margin-left: var(--fs-spacing-1);
+  --fs-price-spot-font-weight: var(--fs-text-weight-bold);
```
Comment (renatamottam) [inline src/components/ui/Price/price.module.scss:12]:
```suggestion
  // Listing
  --fs-price-listing-color           : var(--fs-color-text-light);
  --fs-price-listing-text-decoration : line-through;

  // Spot
  --fs-price-spot-margin-left        : var(--fs-spacing-1);
  --fs-price-spot-font-weight        : var(--fs-text-weight-bold);
```

File: src/components/ui/Price/price.module.scss [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+  // --------------------------------------------------------
+  // Variants Styles
+  // --------------------------------------------------------
+
+  &[data-variant="listing"] {
+    color: var(--fs-price-listing-color);
+    text-decoration-line: var(--fs-price-listing-text-decoration-line);
```
Comment (renatamottam) [inline src/components/ui/Price/price.module.scss:20]:
```suggestion
    text-decoration: var(--fs-price-listing-text-decoration-line);
```

**Review by renatamottam — COMMENTED** [review]
Hey, I left some suggestions :)

File: src/components/ui/Price/Price.stories.mdx [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+
+### Installment
+
+<Canvas>
+  <Story
+    name="installment"
+    args={{
+      variant: 'installment',
+      value: 100,
+      SRText: 'Price:',
```
Comment (ArthurTriis1) [inline src/components/ui/Price/Price.stories.mdx:69]:
```suggestion
```

File: src/components/ui/Price/Price.stories.mdx [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+
+### Listing
+
+<Canvas>
+  <Story
+    name="listing"
+    args={{
+      variant: 'listing',
+      value: 100,
+      SRText: 'Original price:',
```
Comment (ArthurTriis1) [inline src/components/ui/Price/Price.stories.mdx:86]:
```suggestion
```

File: src/components/ui/Price/Price.stories.mdx [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+
+### Selling
+
+<Canvas>
+  <Story
+    name="selling"
+    args={{
+      variant: 'selling',
+      value: 100,
+      SRText: 'Price:',
```
Comment (ArthurTriis1) [inline src/components/ui/Price/Price.stories.mdx:125]:
```suggestion
```

File: src/components/ui/Price/Price.stories.mdx [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+<Meta
+  title="Atoms/Price"
+  component={Price}
+  argTypes={{
+    value: {
+      type: {
+        name: 'number',
+      },
+      description: '`number`',
+    },
```
Comment (ArthurTriis1) [inline src/components/ui/Price/Price.stories.mdx:14]: We can reduce the code using this defaultValues.

```suggestion
      description: '`number`',
      defaultValue: 100,
    },
    SRText: {
      defaultValue: 'Price:'
    }
```

**Review by ArthurTriis1 — APPROVED** [review]
Great! I only left a suggestion.

File: src/components/ui/Price/Price.stories.mdx [hunk 9]
```diff
@@ -0,0 +1,185 @@
...
+  </Story>
+</Canvas>
+
+### Spot
+
+<Canvas>
+  <Story
+    name="spot"
+    args={{
+      variant: 'spot',
```
Comment (hellofanny) [inline src/components/ui/Price/Price.stories.mdx:175]:
```suggestion
      variant: 'spot',
      SRText: 'Sale price:',
```

Actually, I'm not sure 🤔
The main reason why we have this `SRText` here is to help to identify what "type" of price is it when we have more than one price displayed. Maybe we could improve the description for this prop, lemme think.

Ex:
image

Comment (eduardoformiga) [reply src/components/ui/Price/Price.stories.mdx:175]: Agree with you!

**Review by hellofanny — APPROVED** [review]
Niceeeee! 🚀  LGTM!

**Pull request merged**